src/backend/nats.cr

```diff
@@ -0,0 +1,64 @@
+require "nats"
+
+module Cable
+  class NATSBackend < BackendCore
+    register "nats"
+
+    getter nats : NATS::Client do
+      NATS::Client.new(URI.parse(Cable.settings.url))
+    end
+    getter streams = Hash(String, Set(NATS::Subscription)).new { |streams, channel|
+      streams[channel] = Set(NATS::Subscription).new
+    }
+
+    def subscribe_connection
+      nats
+    end
+
+    def publish_connection
+      nats
+    end
+
+    def close_subscribe_connection
+      nats.close rescue nil
+    end
+
+    def close_publish_connection
+      nats.close rescue nil
+    end
```

Comment:
NATS multiplexes everything over a single TCP connection (no connection pool needed), so we ignore any errors caused by closing the client twice.

This _might_ be too naive since `NATS::Client#close` also drains subscriptions (processes any pending messages and synchronous requests), which can also raise errors, so we may want this to be a bit more precise on what exceptions it swallows silently.